pyacq/core/rpc/server.py

```diff
@@ -146,9 +149,20 @@ def __init__(self, addr="tcp://*:*"):
         # Objects that may be retrieved by name using client['obj_name']
         self._namespace = {'self': self}
         
-        # Objects for which we have sent proxies to other machines.
-        self.next_object_id = 0
-        self._proxies = {}  # obj_id: object
+        # Information about objects for which we have sent proxies to other machines.
+        # "object ID" is an integer that uniquely identifies each object that
+        # has been proxied. Multiple requests for the same object will return
+        # proxies with the same object ID. We do _not_ use id(obj) here because
+        # Python may re-use these IDs over time.
+        self._next_object_id = 0  # uniquely identifies proxied objects
+        # "ref ID" is an integer that uniquely identifies a single proxy as it
+        # is sent. Multiple requests for the same object will each have a
+        # different ref ID. These are used for remote reference counting to
+        # ensure that local objects stay alive as long as a remote proxy might
+        # still exist for the object.
+        self._next_ref_id = 0  # uniquely identifies a proxy reference
```

Comment:
Eh.. it's not perfect. I still can't think of a way to make reference management completely automatic across multiple processes.